components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java

```diff
@@ -112,15 +117,24 @@ private Response buildSuccessfulAPIResponse(NotificationResponseBean notificatio
 
         // Check whether detailed api responses are enabled.
         if (isDetailedResponseBodyEnabled()) {
-            SuccessfulUserCreationDTO successfulUserCreationDTO = buildSuccessResponse(notificationResponseBean);
+            String notificationChannel = notificationResponseBean.getNotificationChannel();
+            if (StringUtils.isNotEmpty(notificationChannel) && notificationChannel
```

Comment:
Let's use NotificationChannels.EXTERNAL_CHANNEL.getChannelType().equals(notificationChannel). Then we can get rid of the redundant StringUtils.isNotEmpty(notificationChannel) check